packages/integration_test/android/src/test/java/dev/flutter/plugins/integration_test/FlutterDeviceScreenshotTest.java

```diff
@@ -30,25 +32,32 @@ public void getFlutterView_returnsNullForNonFlutterActivity() {
 
     @Test
     public void getFlutterView_returnsFlutterViewForFlutterActivity() {
-        FlutterView mockFlutterView = mock(FlutterView.class);
-        FlutterActivity mockFlutterActivity = mock(FlutterActivity.class);
-        when(mockFlutterActivity.findViewById(FlutterActivity.FLUTTER_VIEW_ID))
-                .thenReturn(mockFlutterView);
-        assertEquals(
-                FlutterDeviceScreenshot.getFlutterView(mockFlutterActivity),
-                mockFlutterView
-        );
+        // Mock the static call to View.generateViewId that FlutterActivity.FLUTTER_VIEW_ID needs.
+        try (MockedStatic<View> mockedStatic = Mockito.mockStatic(View.class)) {
+            mockedStatic.when(View::generateViewId).thenReturn(123);
+            FlutterView mockFlutterView = mock(FlutterView.class);
+            FlutterActivity mockFlutterActivity = mock(FlutterActivity.class);
+            when(mockFlutterActivity.findViewById(FlutterActivity.FLUTTER_VIEW_ID))
+                    .thenReturn(mockFlutterView);
+            assertEquals(
+                    FlutterDeviceScreenshot.getFlutterView(mockFlutterActivity),
+                    mockFlutterView
+            );
+        }
     }
 
     @Test
     public void getFlutterView_returnsFlutterViewForFlutterFragmentActivity() {
-        FlutterView mockFlutterView = mock(FlutterView.class);
-        FlutterFragmentActivity mockFlutterFragmentActivity = mock(FlutterFragmentActivity.class);
-        when(mockFlutterFragmentActivity.findViewById(FlutterFragment.FLUTTER_VIEW_ID))
-                .thenReturn(mockFlutterView);
-        assertEquals(
-                FlutterDeviceScreenshot.getFlutterView(mockFlutterFragmentActivity),
-                mockFlutterView
-        );
+        // Mock the static call to View.generateViewId that FlutterFragment.FLUTTER_VIEW_ID needs.
```

Comment:
You might mention this issue as to why we dont use robolectric here. Otherwise if I saw this code I might replace it with the inverse of this pr. In general we should rely on robolectric instead of mocking ourselves.

Reply:
Added a link to the pr